### PR TITLE
RemoteDataSources migrated to coroutines.

### PR DESCRIPTION
**Pull request by mshaposhnyk**

### CODE REVIEW

File: app/src/main/java/com/example/internshipgithubclient/network/pull/RestRemotePullDataSource.kt [hunk 1]
```diff
@@ -8,12 +8,16 @@ import com.example.internshipgithubclient.network.toDomain
...
-        return repoApiService.getPullsForRepo(repo.owner.login, repo.name)
-            .subscribeOn(Schedulers.io())
+class RestRemotePullDataSource(private val repoApiService: RepoApiService) : RemotePullDataSource {
+    override suspend fun getAll(repo: Repo): Flow<Pull> {
+        return repoApiService.getPullsForRepo(repo.owner.login, repo.name).asFlow()
+            .map { it.toDomain() }
+            /*.subscribeOn(Schedulers.io())
             .observeOn(AndroidSchedulers.mainThread())
-            .flatMap { list -> Single.just(list.map { it.toDomain() }) }
+            .flatMap { list -> Single.just(list.map { it.toDomain() }) }*/
```
Comment (mnilga) [inline app/src/main/java/com/example/internshipgithubclient/network/pull/RestRemotePullDataSource.kt]: Please remove commented code

File: app/src/main/java/com/example/internshipgithubclient/network/repo/RestRemoteIssueDataSource.kt [hunk 2]
```diff
@@ -4,15 +4,18 @@ import com.example.core.data.RemoteIssueDataSource
...
+class RestRemoteIssueDataSource (private val repoApiService: RepoApiService):RemoteIssueDataSource {
+    override suspend fun getRepoIssues(repo: Repo): Flow<Issue> {
+        return repoApiService.getIssuesForRepo(repo.owner.login, repo.name).asFlow()
+            .map {
+                it.toDomain()
+            }
+            /*.subscribeOn(Schedulers.io())
             .observeOn(AndroidSchedulers.mainThread())
-            .flatMap { list -> Single.just(list.map { it.toDomain() }) }
+            .flatMap { list -> Single.just(list.map { it.toDomain() }) }*/
```
Comment (mnilga) [inline app/src/main/java/com/example/internshipgithubclient/network/repo/RestRemoteIssueDataSource.kt]: Please remove commented code

File: app/src/main/java/com/example/internshipgithubclient/network/repo/RestRemoteRepoDataSource.kt [hunk 3]
```diff
@@ -7,29 +7,40 @@ import com.example.internshipgithubclient.network.toDomain
...
+    override suspend fun getWatchers(repo: Repo): Flow<User> {
+        return repoApiService.getWatchersForRepo(repo.owner.login, repo.name).asFlow()
+            .map {
+                it.toDomain()
+            }.flowOn(Dispatchers.IO)
+        /*.subscribeOn(Schedulers.io())
+        .observeOn(AndroidSchedulers.mainThread())
+        .flatMap {
+            Single.just(it.toDomain())
+        }*/
```
Comment (mnilga) [inline app/src/main/java/com/example/internshipgithubclient/network/repo/RestRemoteRepoDataSource.kt]: Please remove commented code

**Review by mnilga — APPROVED** [review]

File: app/src/main/java/com/example/internshipgithubclient/network/pull/RestRemotePullDataSource.kt (hunk 1 above)
Comment (mshaposhnyk) [reply app/src/main/java/com/example/internshipgithubclient/network/pull/RestRemotePullDataSource.kt]: Removed

File: app/src/main/java/com/example/internshipgithubclient/network/repo/RestRemoteIssueDataSource.kt (hunk 2 above)
Comment (mshaposhnyk) [reply app/src/main/java/com/example/internshipgithubclient/network/repo/RestRemoteIssueDataSource.kt]: Removed

**Pull request merged**